scipy/optimize/tests/test__dual_annealing.py

```diff
@@ -336,3 +337,29 @@ def test_accept_reject_probabilistic(
         rate = 0 if pqv <= 0 else np.exp(np.log(pqv) / (1 - accept_param))
 
         assert_allclose(rate, accept_rate)
+
+    def test_bounds_class(self):
+        # test that result does not depend on the bounds type
+        def func(x):
+            f = np.sum(x * x - 10 * np.cos(2 * np.pi * x)) + 10 * np.size(x)
+            return f
+        lw = [-5.12] * 10
+        up = [5.12] * 10
+
+        # set upper bound to global minimum for a few dimensions
+        up[0] = 0.0
+        up[1] = 0.0
+        up[2] = 0.0
+
+        # run optimizations
+        bounds = Bounds(lw, up)
+        ret_bounds_class = dual_annealing(func, bounds=bounds, seed=1234)
+
+        bounds_old = list(zip(lw, up))
+        ret_bounds_list = dual_annealing(func, bounds=bounds_old, seed=1234)
+
+        # test that found minima, function evaluations and iterations match
+        assert_allclose(ret_bounds_list.x, ret_bounds_class.x, atol=1e-9)
+        assert_allclose(ret_bounds_list.fun, ret_bounds_class.fun, atol=1e-9)
+        assert ret_bounds_list.nfev == ret_bounds_class.nfev
+        assert ret_bounds_list.nit == ret_bounds_class.nit
```

Comment:
This is not a very sensitive test because the maximum number of iterations is being reached. It would be more convincing if the algorithm terminated within the iteration limit. Is there a termination criterion that could be adjusted to make that happen?

Reply:
dual_annealing always runs for the maximum number of iterations, no termination criteria are available. Those asserts for `nfev` and `nit `can be removed then probably.

Reply:
OK. Is it always the same number of function evaluations per iteration?

Reply:
As dual_annealing performs a local search on accepted locations, I guess it does not.

Reply:
OK, so we could probably remove `nit`, but if `nfev` could (in principle) vary if the two ways of specifying bounds were different, then keep it in.

Reply:
Done